### PR TITLE
new OpRandCrop for augmentation

### PR DESCRIPTION
**Pull request by bensha6757**

### CODE REVIEW

File: fuse/data/ops/ops_aug_common.py [hunk 1]
```diff
@@ -179,3 +180,70 @@ def __init__(self, op: OpBase, kwargs_per_step_to_add: Sequence[dict]):
...
+    """
+    Crops a given string based on a randomly generated ratio between 0 and crop_max_ratio.
+
+    Can be used as an augmentation method for cropping a protein/any other input
+    """
+
+    def __init__(
+        self,
+        crop_max_ratio: float = 0.3,
+        crop_both_sides: bool = True,
```
Comment (mosheraboh) [inline fuse/data/ops/ops_aug_common.py:195]: maybe get mode: "left_only", "right_only" or "both"?

**Review by mosheraboh — COMMENTED** [review]
Looks good.
Few questions inline.

File: fuse/data/ops/ops_aug_common.py [hunk 2]
```diff
@@ -179,3 +180,70 @@ def __init__(self, op: OpBase, kwargs_per_step_to_add: Sequence[dict]):
...
+
+        assert 0 <= crop_max_ratio <= 1.0
+        assert any([crop_both_sides, crop_left_only, crop_right_only])
+
+        self.crop_both_sides = crop_both_sides
+        self.crop_left_only = crop_left_only
+        self.crop_right_only = crop_right_only
+        self.crop_max_ratio = crop_max_ratio
+
+    def __call__(
```
Comment (mosheraboh) [inline fuse/data/ops/ops_aug_common.py:220]: what is expected to be stored in key_in.
I assume either tensor or numpy array? mention it.
What are the dims? I assume that the only limitation is that dim 0 is the sequence dim, right?

File: fuse/data/ops/ops_aug_common.py [hunk 3]
```diff
@@ -179,3 +180,70 @@ def __init__(self, op: OpBase, kwargs_per_step_to_add: Sequence[dict]):
...
+        self,
+        sample_dict: NDict,
+        string_key_in: str,
+        string_key_out: str,
+    ) -> dict:
+        input_string = sample_dict[string_key_in]
+        length = len(input_string)
+        crop_ratio = random.uniform(0, self.crop_max_ratio)
```
Comment (mosheraboh) [inline fuse/data/ops/ops_aug_common.py:228]: It's better to use random generator initialized with a given seed.

File: fuse/data/ops/ops_aug_common.py [hunk 4]
```diff
@@ -179,3 +180,70 @@ def __init__(self, op: OpBase, kwargs_per_step_to_add: Sequence[dict]):
...
+
+    def __init__(
+        self,
+        crop_max_ratio: float = 0.3,
+        crop_both_sides: bool = True,
+        crop_left_only: bool = False,
+        crop_right_only: bool = False,
+    ):
+        """
+        :param crop_max_ratio: The maximum ratio of the string length that can be cropped in total.
```
[the remark this answers is not in the document]
Comment (bensha6757) [reply fuse/data/ops/ops_aug_common.py:200]: yes

File: fuse/data/ops/ops_aug_common.py (hunk 2 above)
Comment (bensha6757) [reply fuse/data/ops/ops_aug_common.py:220]: the sample_dict[string_key_in] is a string (the sequence), why do you assume it is a tensor or np array?

Comment (mosheraboh) [reply fuse/data/ops/ops_aug_common.py:223]: My bad, In my mind it was on the already tokenized sequence and the code will work also with tensors or any list like sequence..
Anyway, please add in a comment that you are expecting a sequence represented in string.

**Review by mosheraboh — APPROVED** [review]

**Review by mosheraboh — APPROVED** [review]

**Pull request merged**